extension/cgroupruntimeextension/config.go

```diff
@@ -0,0 +1,28 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cgroupruntimeextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/cgroupruntimeextension"
+
+import "errors" // Config contains the configuration for the cgroup runtime extension.
+
+type Config struct {
+	GoMaxProcs GoMaxProcsConfig `mapstructure:"gomaxprocs"`
+	GoMemLimit GoMemLimitConfig `mapstructure:"gomemlimit"`
+}
+
+type GoMaxProcsConfig struct {
+	Enabled bool `mapstructure:"enabled"`
+}
+
+type GoMemLimitConfig struct {
+	Enabled bool    `mapstructure:"enabled"`
+	Ratio   float64 `mapstructure:"ratio"`
+}
+
+// Validate checks if the extension configuration is valid
+func (cfg *Config) Validate() error {
+	if cfg.GoMemLimit.Ratio < 0 || cfg.GoMemLimit.Ratio > 1 {
+		return errors.New("gomemlimit ratio must be between 0 and 1")
```

Comment:
Should we also exclude `0` and `1`?

Reply:
Good point, I think we should only exclude the `0` value because it is equivalent to disabling this setting and the automemlimit package will return an error: https://github.com/KimMachineGun/automemlimit/blob/main/memlimit/provider.go#L20

The `1` value might be kept because it would mean setting the GOMEMLIMIT to the defined cgroup memory limit. 

Changed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35472/commits/6abcf495a3d6d457cfb9670f1d459fdcdf815fad